xen/arch/arm/coproc/plat/coproc_xxx/coproc_xxc.c

```diff
@@ -0,0 +1,227 @@
+    {
+        .name = "sram",
+        .size = 0,
+        .ops = &vcoproc_xxc_sram_mmio_handler,
+    },
+};
+
+enum COPROC_XXC_IRQ
+{
+    COPROC_XXC_OP = 0,
+    COPROC_XXC_MMU
+};
+
+static struct pcoproc_irq coproc_xxc_irq[] = {
+    {
+        .name = "op",
+        .handler = coproc_xxx_irq_handler,
+    },
+    {
+        .name = "mmu",
+        .handler = coproc_xxx_irq_handler,
+    },
+};
+
+static const struct pcoproc_desc coproc_xxc_desc = {
+    .p_mmio_num = sizeof(coproc_xxc_mmio)/sizeof(struct pcoproc_mmio),
+    .p_mmio = coproc_xxc_mmio,
+    .p_irq_num = sizeof(coproc_xxc_irq)/sizeof(struct pcoproc_irq),
+    .p_irq = coproc_xxc_irq,
+};
+
+static int vcoproc_xxc_op_write(struct vcpu *v, mmio_info_t *info, register_t r,
+                                void *priv)
+{
+    struct vcoproc_mmio *mmio = priv;
+    struct vcoproc_rw_context ctx;
+
+    vcoproc_get_rw_context(v->domain, mmio, info, &ctx);
+    COPROC_DEBUG(ctx.vcoproc->mcoproc->dev,
+                 "domain %u, write r%d=%"PRIregister \
+                 " offset %#08x base %#08x name %s\n",
+                 v->domain->domain_id, ctx.dabt.reg, r, ctx.offset,
+                 (uint32_t)mmio->addr, mmio->m_mmio->p_mmio->name?:"NONAME");
+
+#if 1
+    /* for debug purposes */
+#define COPROC_XXC_OP_INT_REG   0x20
+    if ( ctx.offset == COPROC_XXC_OP_INT_REG )
+    {
+        int i;
+
+        /* Look for MMU irq to inject */
+        for ( i = 0; i < ctx.vcoproc->mcoproc->num_irqs; i++ )
+            if ( ctx.vcoproc->mcoproc->irqs[i].p_irq ==
+                 &coproc_xxc_irq[COPROC_XXC_OP] )
+            {
+                vgic_vcpu_inject_spi(ctx.vcoproc->domain,
+                                 ctx.vcoproc->mcoproc->irqs[i].irq);
+                break;
+            }
+    }
+    else
+        vcoproc_xxx_write(v, info, r, priv);
+#endif
+
+    return 1;
+}
+
+static int vcoproc_xxc_mmu_write(struct vcpu *v, mmio_info_t *info,
+                                 register_t r, void *priv)
+{
+    struct vcoproc_mmio *mmio = priv;
+    struct vcoproc_rw_context ctx;
+
+    vcoproc_get_rw_context(v->domain, mmio, info, &ctx);
+    COPROC_DEBUG(ctx.vcoproc->mcoproc->dev,
+                 "domain %u, write r%d=%"PRIregister
+                 " offset %#08x base %#08x name %s\n",
+                 v->domain->domain_id, ctx.dabt.reg, r, ctx.offset,
+                 (uint32_t)mmio->addr, mmio->m_mmio->p_mmio->name?:"NONAME");
+
+#if 1
+    /* for debug purposes */
+#define COPROC_XXC_MMU_INT_REG   0x30
+    if ( ctx.offset == COPROC_XXC_MMU_INT_REG )
+    {
+        int i;
+
+        /* Look for MMU irq to inject */
+        for ( i = 0; i < ctx.vcoproc->mcoproc->num_irqs; i++ )
+            if ( ctx.vcoproc->mcoproc->irqs[i].p_irq ==
+                 &coproc_xxc_irq[COPROC_XXC_MMU] )
+            {
+                vgic_vcpu_inject_spi(ctx.vcoproc->domain,
+                                 ctx.vcoproc->mcoproc->irqs[i].irq);
+                break;
+            }
```

Comment:
You use almost the same lookup from 2 places (vcoproc_xxc_op_write, vcoproc_xxc_mmu_write). Wouldn't be more cleaner to introduce a helper for such job?